roles/pulp3/tasks/install.yml

```diff
@@ -69,6 +69,13 @@
         - plugin
       when: pulp_source_dir is defined
 
+    - name: Include plugin specific pre tasks
```

Comment:
This still feels weird to include here and that it ought to just be included in the playbook not built into the role.

Reply:
@ehelms I agree, the problem is that the pulp virtualenv does not exist until that step, we even have no access to variables like `pulp_install_dir` in the playbook level before the role is loaded. That is why we need to inject the installation step in that place so it can access the same virtualenv.

Reply:
Here are the 2 variables we need to access, and that vars exists only after the `pulp3` role is loaded.

```yaml
virtualenv: '{{ pulp_install_dir }}'
virtualenv_command: '{{ pulp_python_interpreter }} -m venv'
```

Reply:
The best would be if we do `pip install pulp_rpm|{or path to its source}` and the `setup.py` takes care of building and installing its dependencies, but the documentation says that createrepo_c should be installed manually https://pulp-rpm.readthedocs.io/en/latest/installation.html#install-createrepo-c-from-source

Reply:
Hrmm fair. Still feels like you could bake it into the role such that if installed from source it invokes the tasks laid out vs. having to call to some supplied playbook. That is to say, if someone wants to use the roles to do a source install it should setup all that stuff for them.

Reply:
@ehelms if all the roles was in ansible-galaxy it would be easy to create custom installers for QE. For now I dont know how to setup in that way other than forking, cloning and editing on CI.

Reply:
My interpretation (so correct me where I go astray) of these changes is that when doing a `source` based install, these changes are `required` for Pulp 3 to work. For example, `createrepo` is needed when using the RPM plugin from source.

If the above is true, then IMO the role should contain installing those packages and the setup in these playbooks you are adding since they are essential to the operation of Pulp 3.

Also, you don't need these on galaxy. You can point galaxy at a git repository and include the role paths in your `ansible.cfg`.  See [1]


[1] https://galaxy.ansible.com/docs/contributing/creating_multi.html